presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorNodePartitioningProvider.java

```diff
@@ -71,16 +71,16 @@ public static ConnectorBucketNodeMap createBucketNodeMap(int bucketCount)
         public static ConnectorBucketNodeMap createBucketNodeMap(Map<Integer, Node> bucketToNode)
         {
             requireNonNull(bucketToNode, "bucketToNode is null");
-            if (bucketToNode.isEmpty()) {
-                throw new IllegalArgumentException("bucketToNode is empty");
-            }
-
-            return new ConnectorBucketNodeMap(
-                    bucketToNode.keySet().stream()
-                            .mapToInt(Integer::intValue)
-                            .max()
-                            .getAsInt() + 1,
-                    Optional.of(bucketToNode));
+            int maxBucket = bucketToNode.keySet().stream()
+                    .mapToInt(Integer::intValue)
+                    .peek(bucket -> {
+                        if (bucket < 0) {
```

Comment:
We can also do this check in `FixedBucketNodeMap` constructor. 

A question unrelated to this PR: why Raptor use `Map` to represent bucket to node map instead of an `List`?

Reply:
Good question. Probably just how the data structure was originally built and it leaked into the APIs. Now that you mention it, we should actually remove the map version of the function here, since that should be the responsibility of the caller. I'll add another commit to do that.

Reply:
Actually, that's a bigger change, I'll leave that for a different PR.